src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx

```diff
@@ -1,38 +1,41 @@
-import React, { useState, useEffect } from "react";
+import React, { useEffect } from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
-import { getTokenOwner } from "./../../../services/token";
-import { useEscrowContractUsers } from "../../../hooks";
-import { WrappedDocument } from "@govtechsg/open-attestation";
-
-interface TitleTransferPanelContentProps {
-  document: WrappedDocument;
-}
-
-export const TitleTransferPanelContent = ({ document }: TitleTransferPanelContentProps) => {
-  const [tokenOwnerAddress, setTokenOwnerAddress] = useState("");
-  const { holderAddress, beneficiaryAddress } = useEscrowContractUsers({ escrowContractAddress: tokenOwnerAddress });
-
-  const fetchTokenOwner = async (document: WrappedDocument) => {
-    const address = await getTokenOwner({ document });
-    setTokenOwnerAddress(address);
-  };
+import { useDefaultProvider } from "../../../common/hooks/useDefaultProvider";
+import { useTitleEscrowContract } from "../../../common/hooks/useTitleEscrowContract";
+import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
+import { useContractFunctionHook } from "@govtechsg/ethers-contract-hook";
 
+export const TitleTransferPanelContent = ({ titleEscrow }: { titleEscrow: TitleEscrow }) => {
+  const { call: getBeneficiary, value: beneficiary } = useContractFunctionHook(titleEscrow, "beneficiary");
+  const { call: getHolder, value: holder } = useContractFunctionHook(titleEscrow, "holder");
   useEffect(() => {
-    fetchTokenOwner(document);
-  }, [document]);
-
+    getBeneficiary();
+    getHolder();
+  }, [titleEscrow]); // eslint-disable-line react-hooks/exhaustive-deps
   return (
-    <>
-      <div className="row">
-        <div className="col-12 col-lg">
-          <TitleView role="Beneficiary" address={beneficiaryAddress} />
-          <TitleView role="Holder" address={holderAddress} />
-        </div>
-        <div className="col-12 col-lg">
-          <BLinfo />
-        </div>
+    <div className="row">
+      <div className="col-12 col-lg">
+        <TitleView role="Beneficiary" address={beneficiary} />
+        <TitleView role="Holder" address={holder} />
       </div>
-    </>
+      <div className="col-12 col-lg">
+        <BLinfo />
+      </div>
+    </div>
   );
 };
+interface TitleTransferPanelContentContainerProps {
+  tokenId: string;
+  tokenRegistryAddress: string;
+}
+
+export const TitleTransferPanelContentContainer = ({
+  tokenId,
+  tokenRegistryAddress
+}: TitleTransferPanelContentContainerProps) => {
+  const { provider } = useDefaultProvider(); // Component only need read only access
+  const { titleEscrow } = useTitleEscrowContract(tokenRegistryAddress, tokenId, provider);
+
+  return titleEscrow ? <TitleTransferPanelContent titleEscrow={titleEscrow} /> : null;
```

Comment:
dont want to show a beautiful loader while there is no title escrow ? (and what if there is an error ?)

Reply:
not in scope of refactor. It just shows the gaps in the application design. I should probably file a story / issue?

Reply:
yeah ok open an issue and close this comment 👍